modules/ugdk-core/include/ugdk/action/animationplayer.h

```diff
@@ -31,7 +31,7 @@ class AnimationPlayer : public MediaPlayer {
         frame_change_callback_ = callback;
     }
 
-    const Animation* current_animation() const {
+    std::shared_ptr<const Animation> current_animation() const {
```

Comment:
Actually, I think @kapitenA got carried out here. This PR was supposed to only aply shared_ptr to `AnimationTable` as per #63. Nonetheless, what you said is indeed a problem. We need a custom smart pointer that has only one ownership plus many "weak" references.